lib/clearstream_client/clients/base_client.rb

```diff
@@ -22,6 +22,7 @@ def index
     end
 
     def create
+      @data.delete(:resource)
```

Comment:
What about if we have a `ClearStreamData` acting here as a type that we will know what the data 
 ClearStream needs to have in the request? 

With this, you can pass @data to it and it will return an instance of `ClearStreamData`, and you can also return an error saying: "You must have XYZ keys to make this call". With this, we will make sure we are passing the correct data and someone else that is checking this can make sure all the fields we need to pass to clearstream.

Reply:
@knewter thoughts ?